src/rdb/tests/rdbt.c

```diff
@@ -272,14 +272,6 @@ rdbt_find_leader(crt_group_t *group, uint32_t nranks, uint32_t nreplicas,
 			}
 		}
 
-		if (!resp_isvalid) {
-			printf("ERR: rank %u invalid reply: rc="DF_RC", "
-			       "hint is %s valid (rank=%u, term="DF_U64")\n",
-			       rank, DP_RC(rc_svc), resp_isvalid ? "" : "NOT",
-			       h.sh_rank, h.sh_term);
-			rc = -1;
-			break;
-		}
```

Comment:
why remove this section? may confirm with @liw as well.

Reply:
This is also a dead code case, all condition checks in the for loop will break when !resp_isvalid is true:
if (!resp_isvalid)
                break;

So the removed section will never be reached.

Reply:
@kccain, any comment?

Reply:
Yes @GitHuaKuang appears correct on the dead code issue here. This is my mistake from PR [2708](https://github.com/daos-stack/daos/pull/2708/). I might take all of the preceding if (!resp_isvalid) checks and instead of break out of the loop, goto resp_valid_check instead. "resp_valid_check" would be a new label inserted here at this if (!resp_valid) test.

Should this kind of logic change be done separately in its own PR? In which case I could take that, restructure and manually retest (this test is not yet automated). If it can be handled here, that's fine too - I can manually retest when this lands.

Reply:
There's a lot of cleanups in this PR and it would be good if they could land so I prefer to keep the simple things here, and if in the process we identify other, more complex, areas that need investigation then they should be handled in other PRs.

If you know this code and can test it then it makes more sense for you to take this on so we can move the PR forwards IMO.

Reply:
https://scan8.coverity.com/reports.htm#v42092/p12225/fileInstanceId=36363344&defectInstanceId=11131872&mergedDefectId=303781
I moved this CID to Ken, and will revert the changes from this PR so Ken can handle it separately.

Reply:
I created a PR for this rdbt client program problem https://github.com/daos-stack/daos/pull/3741